libp2p 0.37.0 and other updates

max-privatevoid opened the pull request.

alecbcs commented:
@sug0 As far as I understand it unless a new version of Go is needed for a feature, the value in the `go.mod` file is really just the minimum version of Go needed and doesn't need to be updated. For example here's the libp2p [go.mod file](https://github.com/libp2p/go-libp2p/blob/master/go.mod).

sug0 commented:
Ok, I'm closing this issue then.

max-privatevoid committed:
update go deps

go-libp2p: 0.33.2 -> 0.37.0
go-libp2p-kad-dht: 0.25.2 -> 0.28.0
go-multiaddr: 0.12.3 -> 0.14.0

max-privatevoid committed:
update vendorHash
